libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -374,14 +372,13 @@ def _start_vm(self):
             kernel_cmdline += f' {self.kernel_cmdline}'
 
         # Assemble virt-install command
-        cmd = ['virt-install', '--connect', self.hypervisor_host, '--noautoconsole',
+        cmd = ['virt-install', '--connect', self.hypervisor_host, '--autoconsole', 'text',
             '-n', self.vm_name, '--os-variant', 'rhel-unknown', '--location', location,
             '--memory', '4096', '--vcpus', '2', '--disk', 'size=10',
             '--extra-args', kernel_cmdline]
 
         LOGGER.info('Running: ' + repr(cmd))
-        process = subprocess.run(cmd ,check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-        self.log(process.stdout.decode())
+        self.proc_virtinstall = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
I realize this was not done in the original code, but there should be `self.proc_virtinstall.wait` called somewhere, most probably in the teardown before the VM is attempted to be removed. The issue with not calling wait on ended processes is that the PID is taken by zombie process until `wait` syscall is called.

Another question is if the system is (correctly) powered off at the end of the test, nonetheless, the virt-install process should end after `virsh destroy` is called and once the virt-install process ends it should be safe to call `virt undefine --remove-all-storage` without any sleep and that way prevent race condition.